src/arrayMethodSort.js

```diff
@@ -4,8 +4,36 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+    let tempFunction = compareFunction;
+
+    if (typeof compareFunction !== 'function') {
+      tempFunction = function (a, b) {
+        const strA = String(a);
+        const strB = String(b);
+
+        if (strA < strB) {
+          return -1;
+        }
+
+        if (strA > strB) {
+          return 1;
+        }
+
+        return 0;
+      };
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (tempFunction(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
+    }
```

Comment:
The sorting algorithm used here is a bubble sort, which has a time complexity of O(n^2). While this is fine for small arrays, it can be inefficient for larger ones. Consider implementing a more efficient sorting algorithm like quicksort or mergesort for better performance on larger datasets.

Reply:
I changed to quicksort